File: IT/Strutture dati/Python/sorting.py
# file sorting.py

from __future__ import print_function
from random import randint
from operator import itemgetter

# dichiara una lista di interi casuali

li = [randint(1, 100) for i in range(10)]

print("Lista originale:")
print(li)

# ordina gli elementi della lista

li.sort()

print("Lista ordinata:")
print(li)

# crea una lista di tuple

li2 = [(7, 'K'), (5, 'M'), (3, 'B'), (1, 'R'), (6, 'Y'), (4, 'C'), (2, 'G')]

print("Lista di tuple originale:")
print(li2)

# crea una seconda lista con gli elementi della prima lista, ordinandoli per numero

li3 = sorted(li2, key=itemgetter(0))

print("Lista di tuple ordinate:")
print(li3)

File: IT/1/Containers/Java/Sorting.java
/* file Sorting.java */

import java.util.Random;
import java.util.List;
import java.util.ArrayList;
import java.util.Collections;

public class Sorting {
	public static void main(String [] args) {
		/* dichiara una lista di interi casuali */

		List <Integer> arr = new ArrayList<Integer>();

		Random rand = new Random();

		for (int i = 0; i < 10; i++) {
			int n = rand.nextInt(100) + 1;
			arr.add(n);
		}

		System.out.println("Lista originale:");
		System.out.println(arr);

		/* ordina gli elementi dell'array */

		Collections.sort(arr);

		System.out.println("Lista ordinata:");
		System.out.println(arr);
	}
}

